content/en/platform/corda/5.0-dev-preview-1/tutorials/building-cordapp/c5-basic-cordapp-state-2.md

```diff
@@ -0,0 +1,210 @@
+---
+date: '2021-09-15'
+section_menu: corda-5-dev-preview
+menu:
+  corda-5-dev-preview:
+    identifier: corda-corda-5.0-dev-preview-1-tutorial-c5-basic-cordapp-state-2
+    parent: corda-5-dev-preview-1-tutorials-building-cordapp
+    weight: 1030
+tags:
+- tutorial
+- cordapp
+title: "Write an advanced state"
+---
+
+Now that you have created a state to represent a voucher, you must create a `BoardingTicket` state to represent the ticket that the customer receives when they redeem their `MarsVoucher`. This state has some additional complexity. You will be creating this state in the same directory as `MarsVoucher` (`contracts/src/main/kotlin/net/corda/missionMars/states/`).
+
+## Create the `BoardingTicket` state
+
+The `BoardingTicket` state represents the ticket that Mars Express issues to a customer when they redeem their `MarsVoucher`. The state must include:
+
+* Trip information.
+* The company issuing the ticket. The customer can then verify which company the voucher was purchased from.
+* The current owner of the ticket. The company can then verify the identity of the customer who redeems the voucher.
+* The number of days until the rocket launch. Both parties are aware of the trip date.
+
+The `BoardingTicket` state must be transferable between the issuer and the customer. A customer must redeem their boarding ticket when they get on the rocket to Mars. The value of the ticket does not change until it is redeemed, when the state is consumed and the ticket cannot be used again.
+
+### Set up the `BoardingTicket` class
+
+1. Right-click the **states** folder, select **New > Kotlin Class/File**.
+
+2. In the **New Kotlin Class/File** window, select **Class** and create a file called `BoardingTicket`.
+
+3. Open the file.
+
+### Define the `BoardingTicket` data class
+
+Next, define the `BoardingTicket` data class. Include these variables:
+
+* `description` - Trip information. Use type `String`.
+* `marsExpress` - The space travel company issuing the ticket. Use type `Party`.
+* `owner` - The party exchanging the voucher for the ticket. Use type `Party`.
+* `daysUntilLaunch` - The number of days until the launch date. Use type `int`.
```

Comment:
this is a bit odd, why have an imutable value rep[resent somethign that one would expect to change over time?

Reply:
Good point. I'm not sure.